#DPATCHLEVEL=0

2005-12-29  Soeren D. Schulze  <soeren.d.schulze@gmx.de>

	* i386/i386at/model_dep.c (reboot_on_panic) [!MACH_KBD]: New variable.
	(c_boot_entry) [!MACH_KBD]: Set reboot_on_panic to 0 if kernel_cmdline
	contains '-H'.
	* kern/debug.c (panic): Call halt_all_cpus with reboot_on_panic as
	argument.


--- i386/i386at/model_dep.c	2004-11-29 06:34:51.000000000 +0100
+++ i386/i386at/model_dep.c	2005-12-29 16:01:03.000000000 +0100
@@ -94,6 +94,10 @@
 
 int		rebootflag = 0;	/* exported to kdintr */
 
+#if ! MACH_KBD
+boolean_t reboot_on_panic = 1;
+#endif
+
 /* XX interrupt stack pointer and highwater mark, for locore.S.  */
 vm_offset_t int_stack_top, int_stack_high;
 
@@ -347,6 +351,11 @@
 		cninit();		/* need console for debugger */
 		Debugger();
 	}
+#else
+	if (strstr (kernel_cmdline, "-H "))
+	  {
+	    reboot_on_panic = 0;
+	  }
 #endif	/* MACH_KDB */
 
 	machine_slot[0].is_cpu = TRUE;
--- kern/debug.c	2000-07-23 02:34:12.000000000 +0200
+++ kern/debug.c	2005-12-29 15:21:46.000000000 +0100
@@ -120,6 +120,10 @@
 	}
 }
 
+#if ! MACH_KBD
+extern boolean_t reboot_on_panic;
+#endif
+
 /*VARARGS1*/
 void
 panic(const char *s, ...)
@@ -161,7 +165,7 @@
 #if	MACH_KDB
 	Debugger("panic");
 #else
-	halt_all_cpus (1);
+	halt_all_cpus (reboot_on_panic);
 #endif
 }
 
